idaes/core/surrogate/pysmo/sampling.py

```diff
@@ -1536,3 +1535,199 @@ def sample_points(self):
+            ylabels (list): List of column names (if **data_input** is a dataframe) or column numbers (if **data_input** is an array) for the dependent/output variables. Only used in "selection" mode. Default is None.
+
+        Returns:
+            **self** function containing the input information
+
+        Raises:
+            ValueError: The **data_input** is the wrong type
+
+            ValueError: When a non-implemented distribution is supplied in list_of_distributions
+
+            IndexError: When invalid column names are supplied in **xlabels** or **ylabels**
+
+            Exception: When the **number_of_samples** is invalid (not an integer, too large, zero, negative)
+
+
+        """
+        if sampling_type is None:
+            sampling_type = "creation"
+            self.sampling_type = sampling_type
+            print("Creation-type sampling will be used.")
+        elif not isinstance(sampling_type, str):
+            raise Exception("Invalid sampling type entry. Must be of type <str>.")
+        elif (sampling_type.lower() == "creation") or (
+            sampling_type.lower() == "selection"
+        ):
+            sampling_type = sampling_type.lower()
+            self.sampling_type = sampling_type
+        else:
+            raise Exception(
+                'Invalid sampling type requirement entered. Enter "creation" for sampling from a range or "selection" for selecting samples from a dataset.'
+            )
+        print("Sampling type: ", self.sampling_type, "\n")
+
+        if self.sampling_type == "selection":
+            if isinstance(data_input, (pd.DataFrame, np.ndarray)):
+                self.selection_columns_preprocessing(data_input, xlabels, ylabels)
+            else:
+                raise ValueError(
+                    'Pandas dataframe or numpy array required for sampling_type "selection."'
+                )
+
+            # Catch potential errors in number_of_samples
+            if number_of_samples is None:
+                print(
+                    "\nNo entry for number of samples to be generated. The default value of 5 will be used."
+                )
+                number_of_samples = 5
+            elif number_of_samples > self.data.shape[0]:
+                raise Exception(
+                    "Sample size cannot be greater than number of samples in the input data set"
+                )
+            elif not isinstance(number_of_samples, int):
+                raise Exception("number_of_samples must be an integer.")
+            elif number_of_samples <= 0:
+                raise Exception("number_of_samples must a positive, non-zero integer.")
+            self.number_of_samples = number_of_samples
+
+        elif self.sampling_type == "creation":
+            if not isinstance(data_input, list):
+                raise ValueError(
+                    'List entry of two elements expected for sampling_type "creation."'
+                )
+            elif len(data_input) != 2:
```

Comment:
These three checks all produce the same exception - this would seem to me to be a case of a single `elif` with  a set of `or` statements.

Reply:
Fixed.